libnxz.h

```diff
@@ -50,12 +50,13 @@
 typedef unsigned int uInt;
 
 /* zlib crc32.c and adler32.c */
-extern ulong nx_adler32(ulong adler, const char *buf, uint len);
-extern ulong nx_adler32_z(ulong adler, const char *buf, size_t len);
+extern ulong nx_adler32(ulong adler, const unsigned char *buf, size_t len);
+extern ulong nx_adler32_z(ulong adler, const unsigned char *buf, size_t len);
```

Comment:
Both `nx_adler32` functions have the same parameters.
Unfortunately, `nx_adler32_z` has been around since 0.61 and we can't remove it from the ABI.
However, we should be able to remove it from the API here.

Reply:
Done. I removed it from the API but left the function definition in the `nx_adler32.c`. I also made `nx_adler32` contain the actual function implementation and `nx_adler32_z` is just an alias to it now.